source/index.ts

```diff
@@ -0,0 +1,152 @@
+import process from 'node:process';
+import parseArguments, {type Options as ParserOptions} from 'yargs-parser';
+import camelCaseKeys from 'camelcase-keys';
+import {trimNewlines} from 'trim-newlines';
+import redent from 'redent';
+import {buildOptions} from './options.js';
+import {buildParserOptions} from './parser.js';
+import {validate, checkUnknownFlags, checkMissingRequiredFlags} from './validate.js';
+import type {Options, ParsedOptions, Result, AnyFlags} from './types.js';
+
+const buildResult = <Flags extends AnyFlags = AnyFlags>({pkg: packageJson, ...options}: ParsedOptions, parserOptions: ParserOptions): Result<Flags> => {
+	const argv = parseArguments(options.argv as string[], parserOptions);
+	let help = '';
+
+	if (options.help) {
+		help = trimNewlines((options.help || '').replace(/\t+\n*$/, ''));
+
+		if (help.includes('\n')) {
+			help = redent(help, 2);
+		}
+
+		help = `\n${help}`;
+	}
+
+	if (options.description !== false) {
+		let {description} = options;
+
+		if (description) {
+			description = help ? `\n  ${description}\n` : `\n${description}`;
+			help = `${description}${help}`;
+		}
+	}
+
+	help += '\n';
+
+	const showHelp = (code?: number) => {
+		console.log(help);
+		process.exit(typeof code === 'number' ? code : 2);
+	};
+
+	const showVersion = () => {
+		console.log(options.version);
+		process.exit(0);
+	};
+
+	if (argv._.length === 0 && options.argv.length === 1) {
+		if (argv['version'] === true && options.autoVersion) {
+			showVersion();
+		} else if (argv['help'] === true && options.autoHelp) {
+			showHelp(0); // TODO: why is this 0 and showHelp is 2?
+		}
+	}
```

Comment:
Why does `cli.showHelp()` exit with code 2, but the auto help exits with code 0?

Reply:
Because `cli.showHelp()` is generally used to show the help on command misuse / incorrect command. While normal `cli --help` is not incorrect.

https://github.com/sindresorhus/meow/pull/47#issuecomment-244699098

This could use a code comment though, indeed.